File: debug_gguf.py
import sys
sys.path.insert(0, 'llama.orig/evaluation/bitnet_purepy')
from gguf import GGUF

gguf = GGUF('data/bitnet-model.gguf')
print('Alignment:', gguf.alignment)
print('KV count:', len(gguf.kv))
for k, v in gguf.kv.items():
    if isinstance(v, (int, float, str, bool)):
        print(f'  {k}: {v}')
    else:
        print(f'  {k}: type {type(v).__name__}')
print('Tensor count:', len(gguf.tensors))
for name, info in gguf.tensors.items():
    print(f'  {name}: dims={info["dims"]}, type={info["ggml_type"]}, off={info["data_off"]}')
gguf.close()

File: src/rag/embedding-providers/bitnet/debug-gguf.ts
import { GGUF } from './gguf.js';

async function main() {
  const gguf = await GGUF.open('data/bitnet-model.gguf');
  console.log('KV entries:', gguf.kv.size);
  let count = 0;
  for (const [key, value] of gguf.kv) {
    console.log(`  ${key}: ${value}`);
    if (++count >= 5) break;
  }
  console.log('Tensor count:', gguf.tensors.size);
  let tensorCount = 0;
  for (const [name, info] of gguf.tensors) {
    console.log(`  ${name}: dims=${info.dims}, type=${info.ggmlType}, off=${info.dataOff}`);
    if (++tensorCount >= 5) break;
  }
  await gguf.close();
  console.log('Done');
}

main().catch(err => {
  console.error(err);
  process.exit(1);
});